src/k8s/pkg/k8sd/api/cluster_remove.go

```diff
@@ -32,7 +32,7 @@ func (e *Endpoints) postClusterRemove(s *state.State, r *http.Request) response.
 		defer cancel()
 	}
 
-	log := log.FromContext(s.Context).WithValues("name", req.Name)
+	log := log.FromContext(ctx).WithValues("name", req.Name)
```

Comment:
if microcluster has dropped `s.Context`, let's make a `func (e *Endpoints) Context() context.Context` instead and use that.

Reply:
As discussed offline: This would only be required for the long-running cleanup tasks which should be addressed here https://github.com/canonical/microcluster/pull/193. The remaining uses of `s.Context` should be sufficient with `r.Context`

Reply:
As discussed in MM: the cleanup is now entirely moved to the remove hook. PR is updated to explain the rationale.